src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -29,78 +30,65 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
-                        {
-                            new()
-                            {
-                                Label = "Filter 1"
-                            },
-                            new()
-                            {
-                                Label = "Filter 2",
-                                FilterGroups = new List<FilterGroup>
-                                {
-                                    new()
-                                    {
-                                        Label = "Filter 2 group 1",
-                                    },
-                                    new()
-                                    {
-                                        Label = "Filter 2 group 2",
-                                        FilterItems = new List<FilterItem>
-                                        {
-                                            new()
-                                            {
-                                                Label = "Filter 2 group 2 item 1"
-                                            }
-                                        }
-                                    },
-                                }
-                            }
-                        },
-
-                        IndicatorGroups = new List<IndicatorGroup>
-                        {
-                            new()
-                            {
-                                Label = "Indicator group 1",
-                                Indicators = new List<Indicator>
-                                {
-                                    new()
-                                    {
-                                        Label = "Indicator group 1 item 1"
-                                    }
-                                }
-                            }
-                        },
-                    }
-                }
-            };
-
+            var release = _releaseGenerator.Generate();
+            
+            var subjects = _subjectsGenerator
+                .ForIndex(1, s => s
+                    .SetFilters(_ => _filterGenerator
+                        .WithFilterGroups(_ => _filterGroupGenerator
+                            .WithFilterItems(_ => _filterItemGenerator
+                                .GenerateList(1))
+                            .GenerateList(1))
+                        .GenerateList(3)))
+                .GenerateList(2);
+            
+            var subject2IndicatorGroup = _indicatorGroupGenerator
+                .WithSubject(subjects[1])
+                .WithIndicators(_indicatorGenerator.GenerateList(1))
+                .Generate();
+
+            var releaseSubjects = _releaseSubjectGenerator
+                .ForInstance(s => s.Set(rs => rs.Release, release))
+                .ForInstance(s => s.Set(rs => rs.Subject, (_, _, context) => subjects[context.Index]))
+                .GenerateList(subjects.Count);
```

Comment:
So this is an example of where I want to simplify assigning a variable from a list of possibilities (in this example, a pre-existing `subjects` List) so that we generate one of these ReleaseSubjects per Subject in that List, and assign it the relevant Subject from the List.  I've written about some possibilities of different ways to support this in the PR description, but I wouldn't want to be individually using `ForIndex` if we had, say, 10 Subjects to deal with.

One of my suggestions that I mentioned would be to have an extension method like `WithSubjects(List<Subject>)` on `ReleaseSubjectGenerator` that took that List and distributed them across the generated ReleaseSubjects using the SetterContext.Index.  See PR description for more suggestions.

Reply:
OK so you're saying try to eliminate the need to have a pre-existing `subjects` List in the first place by creating them all in situ here.

Yeah I'm happy to do that - there may be cases where we end up with some pretty deep nesting of data generation if we always stick to this pattern, but I'm happy to give it a go and see what it looks like.

Reply:
Hey @ntsim - see my answer here, thanks: https://github.com/dfe-analytical-services/explore-education-statistics/pull/3889#discussion_r1112101438

Reply:
Done ta - see reply to the link above